pp/go/relabeler/appender/storage.go

```diff
@@ -135,6 +135,7 @@ func (qs *QueryableStorage) write() bool {
 	persisted := make([]string, 0, lenHeads)
 	for _, head := range heads {
 		start := qs.clock.Now()
+		head.MergeOutOfOrderChunks()
```

Comment:
it should be done on finalization, because merge of out of order chunks lead to data mutation, which may conflict with read on query